### PR TITLE
Add steps to release pipeline to publish to azure CDN

### PR DESCRIPTION
**Pull request by tjololo**
## Description

Using the existing release script to upload the files controlled by app-frontend-react to the new cdn solution.
For now both the old and the new CDN should be updated, once we are truly confident in the setup we can remove the old.

The new steps require som secrets, will mark the PR as ready when they are available.

## Related Issue(s)

- relates to: [altinn/altinn-studio#7566](https://github.com/Altinn/altinn-studio/issues/7566)

## Verification

- Manual testing
  - [ ] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added
  - [ ] Cypress E2E test(s) have been added
  - [ ] No automatic tests are needed here
  - [ ] I want someone to help me make some tests
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been updated
  
  - [ ] No changes/updates needed
- Changes/additions to component properties
  - [ ] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [ ] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [ ] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board

### CODE REVIEW

**Review by olemartinorg — APPROVED** [review]
Hmm, a few questions..

Won't this rely on the existing CDN repo being there? Right now it copies the entire `toolkits` folder, which contains all the previous versions as well as the new one. If we pull out the rug by removing the git repo backing from the old CDN, would that mean we would overwrite the `toolkits` folder on azure with one that _only_ contains the newest version? If new files are merged in, that won't become a problem.

**Comment by tjololo** [comment]
Good questions as always, since the azure step leverages copy no files will be deleted, but if a file is updated it will be overridden. The downside is that this will do a full copy of all the files every time we release so I think I will need to change that.
Might need to extend the release.sh with an --azure (or something) argument where the logic to copy only the new files to azure.
This will keep all the release logic inside the release.sh script. If you agree I can make these changes to the script and request a new review after I'm done

**Comment by olemartinorg** [comment]
@tjololo Yup, I agree! Part of the reason I made the `release.sh` script is that it's actually possible to run it locally in order to test out what would happen. That makes changes much easier to verify, so I'm all for adding to that script instead of putting things on the outside.

As an aside, will we at some point remove the git-backing for CDN? Right now the repo is fairly large for a git repo, and we would like to make it a bit larger (by adding source maps), but so far we've held that back in anticipation of the new CDN. Maybe we could get usage reports from azure in the future to make it possible to remove older (unused) versions? It's most likely not very critical we keep multi-year old patch releases.

**Comment by tjololo** [comment]
Updated the release.sh with logic for syncing with azure.
The new functionality still piggybacks on the copy logic against altinn-cdn, when we are confident in the new azure cdn we can replace the logic with scripts that copy all the files to be published to a folder and sync it. Then this release logic would not need to fetch all the old versions just sync the new version.
The sync operation filters out and only uploads folders matching the regexes and compares the md5 hashes to determin if a sync is needed on those files.
The script has three options for running:
1. `--azure-sa-name <name>` set to an existing folder would trigger an rsync to this folder with the files to be synced (this is an best effort simulation
2. `--azure-sa-name <name>` set to a storageaccount name and `--azure-sa-token <sas-token>` with a token for that storage account. As long as `--azure-sync-cdn` is not present it will perform an dry-run and output what files would be updated in the storageaccount. Needs acces to the storage account but will display changes that would actually be made
3. `--azure-sa-name <name> --azure-sa-token <token> --azure-sync-cdn` performes the actual sync against azure

File: .github/scripts/release.sh [hunk 1]
```diff
@@ -26,6 +27,25 @@ while [[ $# -gt 0 ]]; do
       COMMIT=yes
       shift # pop option
       ;;
+    --azure-cdn-uri)
+      AZURE_CDN_URI=$(realpath "$2")
+      shift # pop option
+      shift # pop option
+      ;;
```
Comment (olemartinorg) [inline .github/scripts/release.sh:34]: This is references below, but the variable seems to be unused? Was this left over from a round of cleanup? :thinking:

Comment (tjololo) [reply .github/scripts/release.sh]: Correct, removed

Comment (olemartinorg) [reply .github/scripts/release.sh]: What about the references below? It now says things like `As --azure-cdn-uri flag not defined`, but there is no such flag anymore. Probably best to do another pass over the comments. :pray:

Comment (tjololo) [reply .github/scripts/release.sh]: nice catch, not on my game today it seems 🙈

**Review by olemartinorg — APPROVED** [review]

**Pull request merged**